pkg/client/options.go

```diff
@@ -335,6 +335,20 @@ func (m MatchingFields) ApplyToDeleteAllOf(opts *DeleteAllOfOptions) {
 	m.ApplyToList(&opts.ListOptions)
 }
 
+// MatchingSelector filters the list/delete operation on the given field
+// selector (or index in the case of cached lists).
+type MatchingSelector struct {
```

Comment:
Should just be able to use a type alias here:

```suggestion
type MatchingSelector fields.Selector
```

Reply:
I had the same thought. Turns out `fields.Selector` is an interface, and Go doesn't allow interface types to be used as method receivers.

Reply:
I can make the type:
```go
type MatchingFieldsSelector struct {
	fields.Selector
}
```
to basically give it the same calling UX (not exactly the same of course).

Reply:
ah, yeah, forgot about that.  Good call.

Reply:
maybe add a note